compaction/compaction_manager.cc

```diff
@@ -385,7 +385,7 @@ future<sstables::compaction_result> compaction_task_executor::compact_sstables_a
     sstables::compaction_result res = co_await compact_sstables(std::move(descriptor), cdata, on_replace, std::move(can_purge));
 
     if (should_update_history) {
-        co_await update_history(*_compacting_table, res, cdata);
+        co_await update_history(*_compacting_table, std::move(res), cdata);
     }
 
     co_return res;
```

Comment:
How can you move res if you co_return it here?

How come nothing broke?

Reply:
That is a mistake. The most interesting is, however, the second question. Turns out that I move `res` when `should_update_history` is true and in this particular case, the caller ignores the returned result. So it was just a luck.